### PR TITLE
Add WASM support

### PR DESCRIPTION
**Pull request by swenson**
Moved some functions and constants around so common ones are in `ui_common.js`.

`ui_js.go` Mostly contains a basic call out to `prompt()` to ask for input. It's not perfect, but it works and is simple.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.18.x]
+        go-version: [1.21.x]
```
Comment (dekimsey) [inline .github/workflows/test.yml:7]: Just confirming, we are setting go.mod to `1.20`, but here we are testing with the newest stable intentionally.

Given it's a library and not a cli, I don't think they are required to match. Just wanted to confirm intent and not a typo.

**Review by dekimsey — APPROVED** [review]
LGTM, left a minor question about go versions in use.

File: .github/workflows/test.yml (hunk 1 above)
Comment (swenson) [reply .github/workflows/test.yml:7]: Correct, this is intentional. Updating `go.mod` to >= 1.21.0 means that the library requires that version as a minimum. But, that seems a bit overreaching. So, we keep it back to 1.20 so we aren't forcing users of this library to upgrade.

Comment (swenson) [reply .github/workflows/test.yml:7]: (See: https://tip.golang.org/doc/go1.21#tools for example)

**Comment by swenson** [comment]
Thanks!

**Pull request merged**